src/components/TaskRunDetailsView/TaskRunDetailsView.tsx

```diff
@@ -51,7 +53,14 @@ export const TaskRunDetailsView: React.FC<React.PropsWithChildren<TaskRunDetails
     );
   }
 
-  const applicationName = taskRun.metadata?.labels[PipelineRunLabel.APPLICATION];
+  const applicationName = taskRun?.metadata?.labels[PipelineRunLabel.APPLICATION];
+  const baseURL = `/application-pipeline/workspaces/${workspace}/applications/${applicationName}/taskruns/${taskRunName}`;
+  const { activeTab } = params;
+
+  if (!activeTab && trStatus) {
+    trStatus === runStatus.Succeeded ? navigate(`${baseURL}/details`) : navigate(`${baseURL}/logs`);
+  }
```

Comment:
move the `navigate` call inside a useEffect

Reply:
updated